ide-common/src/main/java/org/digma/intellij/plugin/analytics/Environment.java

```diff
@@ -73,11 +74,32 @@ public String getCurrent() {
 
     @Override
     public void setCurrent(@NotNull String newEnv) {
+        setCurrent(newEnv,true,false,null);
+    }
+
+    @Override
+    public void setCurrent(@NotNull String newEnv,boolean forceChange) {
+        setCurrent(newEnv,true,forceChange,null);
+    }
+
+    @Override
+    public void setCurrent(@Nullable String newEnv,boolean refreshInsightsView,@Nullable Runnable taskToRunAfterChange){
+        setCurrent(newEnv,refreshInsightsView,false,taskToRunAfterChange);
+    }
+
+    @Override
+    public void setCurrent(@Nullable String newEnv,boolean refreshInsightsView,boolean forceChange,@Nullable Runnable taskToRunAfterChange){
 
         Log.log(LOGGER::debug, "Setting current environment , old={},new={}", this.current, newEnv);
 
         //don't change or fire the event if it's the same env.
-        if (Objects.equals(this.current, newEnv) || StringUtils.isEmpty(newEnv) || NO_ENVIRONMENTS_MESSAGE.equals(newEnv)) {
+        if (!forceChange && Objects.equals(this.current, newEnv) || StringUtils.isEmpty(newEnv) || NO_ENVIRONMENTS_MESSAGE.equals(newEnv)) {
```

Comment:
This is a temporary quick fix for a condition we are not prepared yet, still i don't understand what you mean?

Reply:
lets say newEnv is null or NO_ENVIRONMENTS_MESSAGE. and also forceChange is true. it will enter the block.
something does not make sense. the previous conditions had or sign between them, and now you add `!forceChange &&`

maybe you meant : 

```suggestion
        if (!forceChange && (Objects.equals(this.current, newEnv) || StringUtils.isEmpty(newEnv) || NO_ENVIRONMENTS_MESSAGE.equals(newEnv)) ) {
```